### ISSUE TITLE
Release major version 5

### ISSUE THREAD

**Issue by phil-davis** [sabre-io/dav#1415]
which will support PHP 7.4 8.0 8.1 8.2 and have type declarations on parameters and return values etc as far as possible with features available in PHP 7.4.

If possible, increase the phpstan level to 6. That requires adding a lot of parameters and return type detail in PHPdoc (specially of the shape of arrays. I will try adjusting a few things and then report how it goes, and how big a job it seems.

**Comment by phil-davis** [comment]
Blocked by https://github.com/sabre-io/vobject/issues/588 - I need to work on that issue firts and get `sabre/vobject` 5.0.0 released. `sabre/dav` has a dependency on that.

**Comment by phil-davis** [comment]
Status of `sabre` components as at 2022-09-11

Done:
cache - 2.0.0 (2022-08-23)
event - 6.0.0 (2022-08-29)
http - 6.0.0 (2022-08-31)
uri - 2.3.1 (2022-08-30) major version was not needed
xml - 3.0.0 (2022-08-17)

ToDo:
vobject - https://github.com/sabre-io/vobject/issues/588
dav - this issue

**Comment by BinaryKitten** [comment]
hi - can i politely ask as to the status of the updates?

**Comment by phil-davis** [comment]
> hi - can i politely ask as to the status of the updates?

I was making lots of progress, then got going on vobject https://github.com/sabre-io/vobject/pull/591 and it needs "a lot" (tm) of specifying types through the code. I have not got back to it for the last 3 months.

https://github.com/sabre-io/vobject/pull/591#issuecomment-1384864967

**Comment by BinaryKitten** [comment]
> I was making lots of progress, then got going on vobject [sabre-io/vobject#591](https://github.com/sabre-io/vobject/pull/591) and it needs "a lot" (tm) of specifying types through the code. I have not got back to it for the last 3 months.
> 
> [sabre-io/vobject#591 (comment)](https://github.com/sabre-io/vobject/pull/591#issuecomment-1384864967)

Is there anything anyone can do to help?

**Comment by DeepDiver1975** [comment]
Once #1497 is merged we can create a `4.x` branch and move towards 5.x on master .... objections @phil-davis @staabm ?

**Comment by phil-davis** [comment]
yes, get as many little fixes and tweaks released in the 4.* series and then we can branch it, and do real work in master towards v5.

**Comment by staabm** [comment]
sounds good to me

**Comment by staabm** [comment]
@phil-davis if you think its helpful, I can infer parameter/return types with rector for most stuff, I think
